File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, h: List[int]) -> int:
        l,r = 0, len(h) - 1
        maxArea = 0
        while l < r:
            maxArea = max(maxArea, (r - l) * min(h[l],h[r]))
            if h[r] < h[l]:
                r -= 1
            else:
                l += 1
        return maxArea

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int lowest = prices[0];

        for (int p : prices) {
            lowest = Math.min(lowest, p);
            res = Math.max(res, p - lowest);
        }
        return res;
    }
}
